Harvest.Core/Services/UserService.cs

```diff
@@ -61,6 +71,47 @@ public async Task<User> GetCurrentUser()
             }
         }
 
+        public async Task<IEnumerable<string>> GetCurrentRoles()
+        {
+            var user = await GetCurrentUser();
+            var userRoles = await _dbContext.Permissions
+                .Where(p => p.UserId == user.Id)
+                .Select(p => p.Role.Name)
+                .ToArrayAsync();
+            return userRoles;
+        }
+
+        public async Task<bool> HasAccess(string accessCode)
+        {
+            var roles = _roleResolver(accessCode).Concat(_roleResolver(AccessCodes.SystemAccess));
+            var userRoles = await GetCurrentRoles();
+            return userRoles.Any(r => roles.Contains(r));
+        }
+
         public const string IamIdClaimType = "ucdPersonIAMID";
     }
+
+    public static class UserServiceExtensions
```

Comment:
Why are these extensions and not just part of the user service?  Extensions are nice when we don't control the class, like string or date extensions.  don't get the need for them here.

Reply:
I prefer keeping interfaces simple. It seems redundant to declare something in an interface when it can be implemented as a static method against the interface.